store/todos.js

```diff
@@ -1,26 +1,41 @@
 export const state = () => ({
-  list: [
-    {
-      name: 'vuex',
-      done: false,
-    },
-    {
-      name: 'vuex2',
-      done: false,
-    },
-    {
-      name: 'vuex3',
-      done: false,
-    },
-  ],
+  list: [],
+  isLoading: false,
+  isCreating: false,
+  editableTodo: null,
 })
 
+export const getters = {
+  todos: (state) => {
+    return state.list
+  },
+  completedTodos: (state) => {
+    return state.list.filter((todo) => todo.completed)
+  },
+  incompleteTodos: (state) => {
+    return state.list.filter((todo) => !todo.completed)
+  },
+  isTodoCreating(state) {
+    return state.isCreating
+  },
+}
+
 export const mutations = {
-  add(state, text) {
-    state.list.push({
-      text,
-      done: false,
-    })
+  add(state, todo) {
+    const newTodo = {
+      id: new Date().getTime(),
+      title: todo.title,
+      completed: false,
+      createdAt: new Date(),
+      completedAt: null,
+    }
+
+    state.list = [newTodo, ...state.list]
+
+    state.isCreating = false
```

Comment:
Please keep mutations of state variables seperated. For example, we could move the above line to be changed in action using the mutation `setCreate` or something else.

Reply:
Ok, Noted. Thanks

Reply:
Resolved